Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ServerGameNetPortal.cs

```diff
@@ -112,6 +113,7 @@ void OnClientDisconnect(ulong clientId)
                     {
                         m_LobbyServiceFacade.RemovePlayerFromLobbyAsync(playerId, m_LobbyServiceFacade.CurrentUnityLobby.Id, null, null);
                     }
+                    SessionManager<SessionPlayerData>.Instance.DisconnectClient(clientId);

```

Comment:
In all those instances where we're using singleton access pattern to get the SessionManager I'm thinking about using DI for this (even with keeping sessionmanager a singleton on it's own - just binding the .Instance in DI scope would do the trick). Not 100% on this, would appreciate your thoughts.